cmd/devp2p/internal/ethtest/helpers.go

```diff
@@ -242,18 +242,25 @@ func (s *Suite) createSendAndRecvConns(isEth66 bool) (*Conn, *Conn, error) {
 	return sendConn, recvConn, nil
 }
 
+func (c *Conn) readAndServe(chain *Chain, timeout time.Duration) Message {
+	if c.negotiatedProtoVersion == 66 {
+		_, msg := c.readAndServe66(chain, timeout)
+		return msg
+	}
+	return c.readAndServe65(chain, timeout)
+}
+
 // readAndServe serves GetBlockHeaders requests while waiting
 // on another message from the node.
-func (c *Conn) readAndServe(chain *Chain, timeout time.Duration) Message {
+func (c *Conn) readAndServe65(chain *Chain, timeout time.Duration) Message {
 	start := time.Now()
 	for time.Since(start) < timeout {
 		c.SetReadDeadline(time.Now().Add(5 * time.Second))
 		switch msg := c.Read().(type) {
-		case *Ping:
+		case Ping:
```

Comment:
This change looks like it either makes something that didn't work, work, or vice versa. What's that about?

Reply:
Ha, yes it fixes something, but I accidentally fixed it in the wrong method

Reply:
Read66 returns `GetBlockHeaders` instead of `*GetBlockHeaders`which is returned by 65